src/redprl/syntax.sml

```diff
@@ -10,7 +10,20 @@ struct
   type dir = param * param
 
   type label = string
-  structure LabelDict = SplayDict (structure Key = StringOrdered)
+  structure Fields =
+  struct
+    val empty = []
+
+    exception Absent
+    fun lookup (lbl : label) fields =
+      case List.find (fn x => #1 x = lbl) fields of
+        SOME (_, tm) => tm
+      | NONE => raise Absent
+
+    fun remove lbl = List.filter (fn f => #1 f <> lbl)
```

Comment:
At least in SML/NJ, this doesn't typecheck. Replacing this line with the following works:

```
    fun remove lbl = List.filter (fn (lbl', _) => lbl' <> lbl)
```

(Or one could add a type annotation; the problem is "unresolved flex record".)